src/dnx.coreclr.unix/dnx.coreclr.cpp

```diff
@@ -233,6 +237,113 @@ int InvokeDelegate(host_main_fn host_main, int argc, const char** argv)
     return exit_code;
 }
 
+#if defined(PLATFORM_LINUX)
+
+std::string get_os_version()
+{
+    std::vector<std::string> qualifiers { "DISTRIB_ID=", "DISTRIB_RELEASE=" };
+
+    std::ifstream lsb_release;
+    lsb_release.open("/etc/lsb-release", std::ifstream::in);
+    if (lsb_release.is_open())
+    {
+        std::string os_version;
+        for (std::string line; std::getline(lsb_release, line); )
+        {
+            for (auto& qualifier : qualifiers)
+            {
+                if (line.compare(0, qualifier.length(), qualifier) == 0)
+                {
+                    if (os_version.length() > 0)
+                    {
+                        os_version.append(" ");
+                    }
+                    os_version.append(line.substr(qualifier.length()));
+                }
+            }
+        }
+
+        if (os_version.length() == 0)
+        {
+            fprintf(stderr, "Could not find version information. OS version will default to the empty string.\n");
+        }
+
+        return os_version;
+    }
+
+    fprintf(stderr, "Could not open /etc/lsb_release. OS version will default to the empty string.\n");
+    return "";
+}
+#else
+std::string translate_darwin_version(const std::string release)
+{
+    auto dot_position = release.find(".");
+    if (dot_position == std::string::npos)
+    {
+        fprintf(stderr, "Could not determine os version.\n");
+        return "10.1";
+    }
+
+    auto version = stoi(release.substr(0, dot_position));
+    return std::string("10.").append(std::to_string(version - 4));
```

Comment:
Yes. will do. I took it from here: http://stackoverflow.com/questions/11072804/how-do-i-determine-the-os-version-at-runtime-in-os-x-or-ios-without-using-gesta. This table is especially useful:

15.x.x  OS X 10.11.x El Capitan
14.x.x  OS X 10.10.x Yosemite
13.x.x  OS X 10.9.x Mavericks
12.x.x  OS X 10.8.x Mountain Lion
11.x.x  OS X 10.7.x Lion
10.x.x  OS X 10.6.x Snow Leopard
 9.x.x  OS X 10.5.x Leopard
 8.x.x  OS X 10.4.x Tiger
 7.x.x  OS X 10.3.x Panther
 6.x.x  OS X 10.2.x Jaguar
 5.x    OS X 10.1.x Puma